suite-native/module-settings/src/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Select, SelectValue } from '@suite-native/atoms';
+import { CurrencyType, fiatCurrencies as fiatCurrenciesMap } from '@suite-common/suite-config';
+
+import { selectFiatCurrency, setFiatCurrency } from '../slice';
+import { transformFiatCurrencyToSelectItem } from '../utils';
+
+export const CurrencySelector = () => {
+    const selectedFiatCurrency = useSelector(selectFiatCurrency);
+    const dispatch = useDispatch();
+
+    const handleSelectCurrency = (value: SelectValue) => {
+        dispatch(setFiatCurrency(value as CurrencyType));
```

Comment:
Could you add generics to the Select components so that you don't have to always cast the selected value? _(as CurrencyType in this case)_

Reply:
That sounds good, let's try how it will look